src/components/results/loadflow/limit-violation-result.tsx

```diff
@@ -0,0 +1,354 @@
+        return loadFlowCurrentViolationsColumnsDefinition(
+            intl,
+            { onSortChanged, sortConfig },
+            { updateFilter, filterSelector },
+            filterEnums
+        );
+    }, [
+        filterEnums,
+        filterSelector,
+        intl,
+        sortConfig,
+        onSortChanged,
+        updateFilter,
+    ]);
+
+    const loadFlowVoltageViolationsColumns = useMemo(() => {
+        return loadFlowVoltageViolationsColumnsDefinition(
+            intl,
+            { onSortChanged, sortConfig },
+            { updateFilter, filterSelector },
+            filterEnums
+        );
+    }, [
+        filterEnums,
+        filterSelector,
+        intl,
+        sortConfig,
+        onSortChanged,
+        updateFilter,
+    ]);
+
+    const messages = useIntlResultStatusMessages(intl);
+
+    useEffect(() => {
+        initFilters();
+        if (initSort) {
+            initSort(getIdType(tabIndex));
+        }
+    }, [tabIndex, onSortChanged, updateFilter, initFilters, initSort]);
+
+    useEffect(() => {
+        if (result) {
+            fetchLimitViolations(studyUuid, nodeUuid, {
+                sort: {
+                    colKey: FROM_COLUMN_TO_FIELD_LIMIT_VIOLATION_RESULT[
+                        sortConfig.colKey
+                    ],
+                    sortWay: sortConfig.sortWay,
+                },
+                filters: filterSelector,
+            })
+                .then((overloadedEquipments: OverloadedEquipmentFromBack[]) => {
+                    setIsOverloadedEquipmentsReady(true);
+                    const sortedLines = overloadedEquipments.map(
+                        (overloadedEquipment) =>
+                            makeData(overloadedEquipment, intl)
+                    );
+                    setOverloadedEquipments(sortedLines);
+                })
+                .catch((error) => {
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: 'ErrFetchViolationsMsg',
+                    });
+                })
+                .finally(() => {
+                    setIsOverloadedEquipmentsReady(true);
+                    setIsFetchComplete(true);
+                });
+        }
+    }, [
+        studyUuid,
+        nodeUuid,
+        intl,
+        result,
+        filterSelector,
+        sortConfig,
+        snackError,
+        initFilters,
+        initSort,
+    ]);
+
+    const getRowStyle = useCallback(
+        (params: RowClassParams) => {
+            if (params?.data?.elementId) {
+                return {
+                    backgroundColor: theme.selectedRow.background,
+                };
+            }
+        },
+        [theme.selectedRow.background]
+    );
+
+    const onGridReady = useCallback(({ api }: GridReadyEvent) => {
+        api?.sizeColumnsToFit();
+    }, []);
+
+    const currentViolations = overloadedEquipments.filter(
```

Comment:
You can do this filtering in the backend to avoid to get all rows and filter them in the front depending on the showed tab

Reply:
Done